server/src/test/java/org/opensearch/cluster/routing/OperationRoutingTests.java

```diff
@@ -759,6 +761,111 @@ public void testAdaptiveReplicaSelectionWithZoneAwarenessIgnored() throws Except
         terminate(threadPool);
     }
 
+    public void testWRR() throws Exception {
+        final int numIndices = 2;
```

Comment:
Such a big method, it's hard to understand in one go.

Reply:
Added comments to make it easy to understand. Tried to move common functionalities to a different method